src/geocel/rasterize/Image.cc

```diff
@@ -0,0 +1,118 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file geocel/rasterize/Image.cc
+//---------------------------------------------------------------------------//
+#include "Image.hh"
+
+#include <cmath>
+
+#include "corecel/cont/ArrayIO.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/data/CollectionAlgorithms.hh"
+#include "corecel/io/Repr.hh"
+#include "corecel/math/ArrayOperators.hh"
+#include "corecel/math/ArraySoftUnit.hh"
+#include "corecel/math/ArrayUtils.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with image properties.
+ *
+ * All inputs should be in the native unit system.
+ */
+ImageParams::ImageParams(ImageInput const& inp)
+{
+    CELER_VALIDATE(ArraySoftUnit{real_type{0.001}}(inp.rightward),
+                   << "rightward axis " << repr(inp.rightward)
+                   << " is not a unit vector");
+    CELER_VALIDATE(inp.lower_left != inp.upper_right,
+                   << "lower left corner " << repr(inp.lower_left)
+                   << " and upper right corner cannot be the same");
+    CELER_VALIDATE(inp.vertical_pixels > 0,
+                   << "number of pixels must be positive");
+    CELER_VALIDATE(inp.horizontal_divisor > 0,
+                   << "number of horizontal chunks must be positive");
+
+    ImageParamsScalars scalars;
+
+    // Normalize rightward axis
+    scalars.right = make_unit_vector(inp.rightward);
+
+    // Vector pointing toward the upper right from the lower left corner
+    Real3 diagonal = inp.upper_right - inp.lower_left;
+
+    // Set downward axis to the diagonal with the rightward component
```

Comment:
I think it would be helpful to document this method more. I don't understand why scalars.down should calculated that way that it is.